tensorly/decomposition/tests/test_parafac2.py

```diff
@@ -518,3 +518,57 @@ def test_update_imputed():
     )
 
     assert_allclose(imputed_tensor, tensor)
+
+
+@pytest.mark.parametrize("linesearch", [False, True])
+def test_parafac2_em(linesearch):
+    rng = tl.check_random_state(1234)
+    rank = 3
+
+    # Create a random tensor
+
+    random_parafac2_tensor = random_parafac2(
+        shapes=[(15 + rng.randint(5), 30) for _ in range(25)],
+        rank=rank,
+        random_state=rng,
+    )
+
+    random_parafac2_tensor.factors[0] = random_parafac2_tensor.factors[0] + 0.01
+
+    slices = parafac2_to_slices(random_parafac2_tensor)
+
+    # Form the full data and a mask with ~10% missing values
+
+    slice_mask = [rng.binomial(1, 0.9, size=T.shape(slice)) for slice in slices]
+
```

Comment:
It would be nice to apply the slice mask to the data as well so the test would fail if we completely ignored the mask

```suggestion

    # Apply the mask, setting missing values to zero
    slices = [tl.where(slice_mask, slice, 0) for slice in slices]

```

Reply:
Added that on ce46c43d022536ce871e4f6dde02e4a823da79cb. Thanks a lot for the nice feedback!